cl/api/views.py

```diff
@@ -163,6 +164,64 @@ def coverage_data(request, version, court):
     )
 
 
+def fetch_start_end_dates_for_court(court_id: str, court_name: str):
+    """Fetch start and end dates for court
+
+    :param court_id: Court ID
+    :param court_name: The Name of the Court we use as a label
+    :return: Timechart data formatted for the court
+    """
+    dates = OpinionCluster.objects.filter(docket__court=court_id).order_by(
+        "date_filed"
+    )
+    if dates:
+        return {
+            "id": court_id,
+            "label": court_name,
+            "data": [
+                {
+                    "val": court_id,
+                    "timeRange": [
+                        dates.first().date_filed,
+                        dates.last().date_filed,
+                    ],
+                }
+            ],
+        }
+
+
+def coverage_data_opinions(request: HttpRequest):
+    """Generate Coverage Chart Data
+
+    Accept Post to query court data for timelines-chart on coverage page
+
+    :param request:
+    :return: TimeChart data if any
+    """
+
+    if request.method == "GET":
+        chart_json: dict[str, object] = defaultdict(dict)
+        query_parameters = request.GET.items()
+        data = []
+        if query_parameters:
+            for court_name, value in query_parameters:
+                group, court_id = value.split("_")  # type: ignore
+
+                if group in dict(Court.JURISDICTIONS).keys():
+                    group = dict(Court.JURISDICTIONS)[group]
+
+                court_data = fetch_start_end_dates_for_court(
+                    court_id=court_id, court_name=court_name
+                )
+                if court_data:
+                    chart_json["data"].setdefault(group, []).append(court_data)  # type: ignore
+
+            for key, value in chart_json["data"].items():  # type: ignore
+                data.append({"group": key, "data": value})
```

Comment:
we could avoid the second loop but not without tradeoffs.  The format that is required by the timelines chart is not conducive to easy lookups because it's an array of data that would need to be looped to- to figure out where to stash it.